### PR TITLE
Integrate Safe app sdk

### PR DESCRIPTION
**Pull request by anxolin**
Integrates app with `safe-apps-react-sdk`so it can be embedded in Gnosis Safe UI.

Part of #43 

This PR adds support for local development of the safe-app, and also for PRaul (so you can copy paste the URL of PRAUL)

## Test locally
Run
```bash
yarn start:safe-swap
```

Check in the browser https://localhost:8080/ (notice the S in the HTTPS)
You might need to tell your browser you trust the self-signed certificate.
You should see sth like:
image

## Test local version in Gnosis Safe
Now you can go to https://rinkeby.gnosis-safe.io and add the app

It should look like this
image

## Test using PRaul
You can also add the PRaul instead of your localhost URL. Just copy it from the comment. 

 **NOTE:** 🚨 not currently working as smooth as it should🚨 
 * I'll review this part, but for now add https://pr278--gpui.review.gnosisdev.com
 * Then, navigate to https://rinkeby.gnosis-safe.io/app/#/safes/0xd9395aeE9141a3Efeb6d16057c8f67fBE296734c/apps?appUrl=https://pr278--gpui.review.gnosisdev.com/safe.html
 * EDIT: Ok, I'll fix this in a different PR, it's not that trivial. For now, I think this one is a good start already. #284

image

### CODE REVIEW

**Comment by anxolin** [comment]
@mergifyio rebase

File: getWebpackConfig.js [hunk 1]
```diff
@@ -38,6 +39,86 @@ function _getHtmlPlugin({ app, templatePath, isProduction }) {
...
+
+  // Html Plugin: Generate one entry point HTML page per app
+  const htmlPlugins = apps.map((app) =>
+    _getHtmlPlugin({
+      app,
+      templatePath: config.templatePath,
+      isProduction,
+    }),
+  )
+  const plugins = [...htmlPlugins]
```
Comment (W3stside) [inline getWebpackConfig.js:53]:
```suggestion
  const plugins = htmlPlugins
```
you're not using any others

File: getWebpackConfig.js [hunk 2]
```diff
@@ -38,6 +39,86 @@ function _getHtmlPlugin({ app, templatePath, isProduction }) {
...
+    new FaviconsWebpackPlugin({
+      logo: config.logoPath,
+      mode: 'webapp', // optional can be 'webapp' or 'light' - 'webapp' by default
+      devMode: 'webapp', // optional can be 'webapp' or 'light' - 'light' by default
+      favicons: {
+        appName: appTitle,
+        appDescription: appTitle,
+        developerName: appTitle,
+        developerURL: null, // prevent retrieving from the nearest package.json
+        background: '#dfe6ef',
```
Comment (W3stside) [inline getWebpackConfig.js:66]: can you make these constants at the top of the page?

**Comment by anxolin** [comment]
> can you make these constants at the top of the page?

Let's leave this for the end. I'm still not decide on how to improve the PRaul and make the manifest better. This is a WIP https://github.com/gnosis/gp-ui/pull/284 so, I'll be dealing with it in another PR

**Pull request merged**